src/queue.hpp

```diff
@@ -0,0 +1,120 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#pragma once
+
+#include <atomic>
```

Comment:
This will be used by example custom nodes. We need to update documentation to enforce users of installing compiler with cpp standard that supports atomic, condition_variable, etc.
We might also need adjust building to match the dependencies of current example custom nodes.

Reply:
There is separate task to update the docs after the feature is complete.